xbmc/rendering/RenderSystem.cpp

```diff
@@ -57,7 +58,10 @@ bool CRenderSystemBase::SupportsStereo(RENDER_STEREO_MODE mode) const
 
 void CRenderSystemBase::ShowSplash(const std::string& message)
 {
-  if (!CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_splashImage && !(m_splashImage || !message.empty()))
+  if ((!CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_splashImage ||
+       !CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+           CSettings::SETTING_LOOKANDFEEL_SHOWSPLASH)) &&
+      !(m_splashImage || !message.empty()))
```

Comment:
@joseluismarti You've done a good job so you can ignore this style comment in this PR (unless you want to change it), but long if statements like this can be made much more readable by splitting into multiple lines, with separate early returns, or also using temporary bools to build up the logic used to decide on an early return.

I'll have a stab at this statement, but it's untested and I could be wrong. Still, you see how it improves readability:

```c++
bool disableSplash = true;

if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_splashImage)
 disableSplash = false;

if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
           CSettings::SETTING_LOOKANDFEEL_SHOWSPLASH))
  disableSplash = false;

if (m_splashImage)
  disableSplash = false;

if (!message.empty())
  disableSplash = false;

if (disableSplash)
  return;
```

Reply:
It's true that it would be more readable, but I just add an ```OR``` condition and the original code is working fine.
It would be wiser not to change it at the moment.